### ISSUE TITLE
Editor for .tern-project

### ISSUE THREAD

**Issue by angelozerr** [angelozerr/tern.java#358]
Develop an editor with multi page like for MANIFEST.MF for .tern-project

We could have several tabs like:

 * Overview to choose ECMASCript + Modules + Repository (that we will able to import it)
 * Server which display tern server type, debug mode, etc
 * Script paths 
 * .tern-project : which display content of .tern-project. Once WTP JSON Editor will be released we could benefit with syntax coloration, completion for modules, validations for JSON (syntax + schema).

Any feedback are welcome!

**Comment by paulvi** [comment]
I thought we could use JSON editor with some completion based on metadata (as your JSON Editor defines that)

There's no need for more GUI, as there are already preferences and projects properties pages.

**Comment by angelozerr** [comment]
> I thought we could use JSON editor with some completion based on metadata (as your JSON Editor defines that)

Yes It will work like that, but I have give my code to WTP and I would like to wait that JSON Editor is released to use it inside tern.java. For the moment I would like to manage UI like PDE (Overview, etc).

> There's no need for more GUI, as there are already preferences and projects properties pages.

In my side, I prefer updating .tern-project directly instead of open project properties. So completion, etc will help me. More I would like to add new action like "Import Repository" in the workspace. So I would like to add this action in the editor toolbar (on the top) to avoid searching this action in the menu (I will add it too).

My idea with this .tern-project editor is to provide a Dashboard kind for the tern project..

In some case, I think it's better to have an editor compar eto project properties. For instance for your case with .jshintrc config file, the tern-project editor will able to provide a link to the .jshintrc. You could click on this link to open the jshintrc file in an editor.

**Comment by paulvi** [comment]
![](https://github.com/angelozerr/tern.java/wiki/images/EclipseIDE_TernProjectEditor.png)

Is current version better than JSDT JSON editing support?

**Comment by angelozerr** [comment]
> Is current version better than JSDT JSON editing support?

Today TernprojectEditor extends FormEditor (like PDE) and add a simple Text Editor. So it's very simply. But now there is an editor linked to .tern-project.
